### PR TITLE
refactor: prepare esm helper

### PR DESCRIPTION
**Pull request by viceice**
## Changes:
- add `expose.cjs` file to (dynamic) import `package.json` and `re2`;
- I don't like to make the `regEx` async to use dynamic typescript import.
- We can't import json files in esm by default yet, needs node [`--experimental-json-modules`](https://nodejs.org/api/esm.html#no-json-module-loading)

## Context:
- #13181 

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/expose.cjs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+ * @type {import('./types').RenovatPackageJson}
+ */
+const pkg = (() => require('../package.json'))();
+
+/**
+ * return's re2
+ * @returns {RegExpConstructor}
+ */
+function re2() {
+  return require('re2');
```
Comment (viceice) [inline lib/expose.cjs:14]: We have over 800 references to `regEx`, so i don't like to make it async. so this is a esm workaround, as this file is always interpreted as cjs. 🙃

File: lib/expose.cjs [hunk 2]
```diff
@@ -0,0 +1,17 @@
+// https://stackoverflow.com/a/46745166/10109857
+
+/**
+ * returns renovates package.json
+ * @type {import('./types').RenovatPackageJson}
+ */
+const pkg = (() => require('../package.json'))();
```
Comment (viceice) [inline lib/expose.cjs:7]: require can't used in esm, so this workaround

File: lib/globals.d.ts [hunk 3]
```diff
@@ -12,8 +12,8 @@ declare interface Error {
 // can't use `resolveJsonModule` because it will copy json files and change dist path
 
 declare module '*/package.json' {
-  import { PackageJson } from 'type-fest';
-  const value: PackageJson & { 'engines-next': Record<string, string> };
+  type RenovatPackageJson = import('./types').RenovatPackageJson;
+  const value: RenovatPackageJson;
```
Comment (rarkins) [inline lib/globals.d.ts:16]:
```suggestion
  type RenovatePackageJson = import('./types').RenovatePackageJson;
  const value: RenovatePackageJson;
```

File: lib/types/base.ts [hunk 4]
```diff
@@ -1,4 +1,10 @@
+import type { PackageJson } from 'type-fest';
+
 export interface ModuleApi {
   displayName?: string;
   url?: string;
 }
+
+export type RenovatPackageJson = PackageJson & {
```
Comment (rarkins) [inline lib/types/base.ts:8]:
```suggestion
export type RenovatePackageJson = PackageJson & {
```

**Review by rarkins — CHANGES_REQUESTED** [review]

**Review by rarkins — APPROVED** [review]

**Pull request merged**